app/controllers/users/sessions_controller.rb

```diff
@@ -13,13 +13,21 @@ def new
 
   def create
     @theme = "minimal"
+    Rails.logger.info "Params: #{params[:user].inspect}"
 
     respond_to do |format|
-      format.json {
-        self.resource = warden.authenticate!(scope: resource_name, recall: "#{controller_path}#new")
-        sign_in(resource_name, resource)
-        render status: 200, json: {status: "Success"}
-      }
+      format.json do
+        email = params[:user][:email]
+        password = params[:user][:password]
+        otp_attempt = params[:user][:otp_attempt]
+
+        user = User.find_for_authentication(email: email)
```

Comment:
I assume this `find_for_authentication` method is a devise thing... I'm not familiar with it. 

Stylitically, I might gather parameters (email, password, otp_attempt, and find the user) and do the 'work' outside of the format.json block (closer to the top of the method), and then depending the result of the work, format.json response accordingly....